src/main/java/com/github/msemys/esjc/Settings.java

```diff
@@ -569,6 +589,10 @@ public Settings build() {
                 failOnNoServerResponse = false;
             }
 
+            if (disconnectOnTcpChannelError == null) {
+                disconnectOnTcpChannelError = true;
```

Comment:
Lets use `false` as default. Currently I don't see the benefit of closing connection when client encounters an error...

```suggestion
                disconnectOnTcpChannelError = false;
```

Reply:
I'm very happy to use `false` as default - I just used the existing behaviour as default, but I agree that changing it to `false` would be more useful. I'll update the Javadoc as, well.